src/app/sidenav/sidenav.component.ts

```diff
@@ -24,9 +28,40 @@ export class SidenavComponent implements OnInit {
   constructor() { }
 
   ngOnInit() {
+    this.communities = new Array();
+    this.communitiesFavorited = new Array();
+    // populate 20 random communities
+    for (let i = 0; i < 20; i++) {
+      this.communities.push({ communityname: this.communityName = 'Community ' + i, 
```

Comment:
`this.communityName`  assignment seems useless here, it isn't used anywhere. You can just do `{ communityname:  'Community ' + i, etc...`